src/libraries/Microsoft.NETCore.Platforms/tests/Microsoft.NETCore.Platforms.Tests.csproj

```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);net472</TargetFrameworks>
+    <!-- Don't build for NETFramework during source-build. -->
+    <TargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true'">$(NetCoreAppCurrent)</TargetFrameworks>
```

Comment:
Fixes:
```
Microsoft.NETCore.Platforms.Tests.csproj : error NU1201: Project Microsoft.NETCore.Platforms_temp is not compatible with net472 (.NETFramework,Version=v4.7.2). Project Microsoft.NETCore.Platforms_temp supports: net8.0 (.NETCoreApp,Version=v8.0)
```

Reply:
Why do we build tests during source-build? If you want to make this work without the switch, replace `net472` with `$(NetFrameworkToolCurrent)` in the TargetFrameworks string. `$(NetFrameworkToolCurrent)` is trimmed out during source-build.

The TFMs should probably be in sync anyway.

Reply:
> Why do we build tests during source-build?

We don't build tests during source-build.

Because source-build builds runtime different than Microsoft, it's interesting to see what happens when we run tests under the source-build configuration.

This PR has the changes needed to make this command work:
```
./build.sh --runtimeconfiguration Release --librariesConfiguration Debug --subset clr+libs+libs.tests --test /p:DotNetBuildFromSource=true /p:RuntimeOS=linux
```

https://github.com/dotnet/runtime/issues/81546 is a bug that the tests uncovered that happens only under the source-build configuration.